### ISSUE TITLE
Auto-cache imported scripts

### ISSUE THREAD

**Issue by jakearchibald** [slightlyoff/ServiceWorker#908]
I'm talkin' ES imports, not `importScripts`.

In the same way we auto-cache `importScripts` during service worker install, we should do the same for ES imports.

I imagine this is easier said than done spec-wise, but if modules are in v1, this should be too.

**Comment by mkruisselbrink** [comment]
Fortunately modules aren't in v1?

**Comment by mkruisselbrink** [comment]
But more helpfully, yes, there are a number of issues with the way module service workers are currently specified where don't integrate well with the way module workers are specified in HTML. Caching the module map is only one of those. More in general we'll need to figure out how to deal with update checks and things like that for module workers, as all that algorithm language currently really doesn't make any sense for module workers.

**Comment by jakearchibald** [comment]
Stupid progress breaking all our shit. Cool, v2 then.

**Comment by monica-ch** [comment]
Bug triage: We've discussed, next steps is to investigate around module and implement and then handle spec.

**Comment by yoshisatoyanagisawa** [comment]
tl;dr we can close this as already fixed.

I did further more investigation.
A browser keeps ServiceWorkerRegistrations persistently.
https://w3c.github.io/ServiceWorker/#service-worker-registration-lifetime

ServiceWorkerRegistrations has set of ServiceWorkers
https://w3c.github.io/ServiceWorker/#dfn-service-worker-registration

ServiceWorker has an associated global object
https://w3c.github.io/ServiceWorker/#dfn-service-worker-global-object

The global object has an associated module map.
https://w3c.github.io/ServiceWorker/#serviceworkerglobalscope
https://html.spec.whatwg.org/multipage/workers.html#concept-workerglobalscope-module-map

Since the module map keeps imported scripts, module scripts should also be auto cached.
https://html.spec.whatwg.org/multipage/webappapis.html#module-map

**Comment by monica-ch** [comment]
@yoshisatoyanagisawa Thanks for looking into this, closing as fixed
